pallets/api/src/messaging/benchmarking.rs

```diff
@@ -0,0 +1,447 @@
+//! Benchmarking setup for pallet_api::messaging
+#![cfg(feature = "runtime-benchmarks")]
+
+use ::ismp::{
+	messaging::hash_request,
+	module::IsmpModule,
+	router::{Request, Response as IsmpResponse, Timeout},
+};
+use ::xcm::latest::{Junctions, Location};
+use frame_benchmarking::{account, v2::*};
+use frame_support::{
+	dispatch::RawOrigin,
+	traits::{Currency, EnsureOrigin},
+	BoundedVec,
+};
+use sp_core::{Get, H256};
+use sp_io::hashing::blake2_256;
+use sp_runtime::traits::One;
+use sp_std::vec;
+
+use super::*;
+use crate::messaging::test_utils::*;
+
+const SEED: u32 = 1;
+type RuntimeOrigin<T> = <T as frame_system::Config>::RuntimeOrigin;
+
+// See if `generic_event` has been emitted.
+fn assert_has_event<T: Config>(generic_event: <T as crate::messaging::Config>::RuntimeEvent) {
+	frame_system::Pallet::<T>::assert_has_event(generic_event.into());
+}
+
+#[benchmarks(
+	where
+	T: pallet_balances::Config + pallet_timestamp::Config,
+)]
+mod messaging_benchmarks {
+	use super::*;
+
+	/// # Parameters
+	/// - `x`: `Linear<1, { T::MaxRemovals::get() }>`   The number of message removals to perform
+	///   (bounded by `MaxRemovals`).
+	#[benchmark]
+	fn remove(x: Linear<1, { T::MaxRemovals::get() }>) {
```

Comment:
No xcm message consideration, but it seemingly is down to ismp storage perhaps being the worst case.